scripts/docker-entrypoint.sh

```diff
@@ -60,17 +65,96 @@ if [ "$JDBC_CLASS" == "" ]; then
+fi
+
+if [ -e "$custom_artifactory_apikey_file" ]; then
+   export SECRET_CUSTOM_ARTIFACTORY_APIKEY=$(<${custom_artifactory_apikey_file})
+else
+   export SECRET_CUSTOM_ARTIFACTORY_APIKEY=${CUSTOM_ARTIFACTORY_APIKEY}
+fi
+
+custom_artifactory_auth=""
+if [ "$SECRET_CUSTOM_ARTIFACTORY_USERNAME" != "" ] || [ "$SECRET_CUSTOM_ARTIFACTORY_PASSWORD" != "" ]; then
+    if [ "$SECRET_CUSTOM_ARTIFACTORY_USERNAME" == "" ] || [ "$SECRET_CUSTOM_ARTIFACTORY_PASSWORD" == "" ]; then
+        echo "SECRET_CUSTOM_ARTIFACTORY_USERNAME & SECRET_CUSTOM_ARTIFACTORY_PASSWORD must be specified for basic authentication for custom artifactory."
+        exit 1
+    else
+        echo "Using basic authentication for custom artifactory to download JDBC driver."
+        custom_artifactory_auth="-u "$SECRET_CUSTOM_ARTIFACTORY_USERNAME":"$SECRET_CUSTOM_ARTIFACTORY_PASSWORD
+    fi
+fi
+
+if [[ "$custom_artifactory_auth" == "" && ( "$SECRET_CUSTOM_ARTIFACTORY_APIKEY_HEADER" != "" || "$SECRET_CUSTOM_ARTIFACTORY_APIKEY" != "" ) ]]; then
+    if [ "$SECRET_CUSTOM_ARTIFACTORY_APIKEY_HEADER" == "" ] || [ "$SECRET_CUSTOM_ARTIFACTORY_APIKEY" == "" ]; then
+        echo "SECRET_CUSTOM_ARTIFACTORY_APIKEY_HEADER & SECRET_CUSTOM_ARTIFACTORY_APIKEY must be specified for authentication using api key for custom artifactory."
+        exit 1
+    else
+        echo "Using API key for authentication of custom artifactory to download JDBC driver."
+        custom_artifactory_auth="-H "$SECRET_CUSTOM_ARTIFACTORY_APIKEY_HEADER":"$SECRET_CUSTOM_ARTIFACTORY_APIKEY
+    fi
+fi
+
+custom_artifactory_certificate=''
+if [ "$(ls -A "${pega_root}/artifactory/cert"/*)" ]; then
+     if [ "$(ls "${pega_root}/artifactory/cert"/* | wc -l)" == 1 ]; then
+       echo "Certificate is provided for custom artifactory's domain ssl verification."
+       # get certificate name
+       for certfile in "${pega_root}/artifactory/cert"/*
+        do
+           echo "folder name: ${pega_root}/artifactory/cert"
+           filename=$(basename "$certfile")
+           ext="${filename##*.}"
+           echo "$filename"
+           if [ "$ext" = "cer" ] || [ "$ext" = "pem" ] || [ "$ext" = "crt" ] || [ "$ext" = "der" ]; then
+              echo "$certfile"
+              custom_artifactory_certificate="--cacert "$certfile
+           else
+              echo "curl needs valid format certificate file for ssl verification."
+              exit 1
+           fi
+        done
+     else
+       echo "Provide one certificate file. The file may contain multiple CA certificates."
+       exit 1
+     fi
+fi
+
 if [ "$JDBC_DRIVER_URI" != "" ]; then
+  curl_cmd_options=''
+  if [ "$ENABLE_CUSTOM_ARTIFACTORY_SSL_VERIFICATION" == true ]; then
+    echo "Establishing a secure connection to download driver."
+    curl_cmd_options="curl -sSL $custom_artifactory_auth $custom_artifactory_certificate"
+  else
+    echo "Establishing an insecure connection to download driver."
+    curl_cmd_options="curl -ksSL $custom_artifactory_auth"
```

Comment:
Now that we can import certs prior to the curl call, we should drop support for the -k flag.

Reply:
@petejo From the pega-helm-charts ENABLE_CUSTOM_ARTIFACTORY_SSL_VERIFICATION is always defaulted to true, in which case we try to make only secure connections.
In cases where client doesn't want to provide a certificate or certificate is expired, one can use this option to skip SSL verification.  Considering above cases, we have added this option where client can disable SSL verification, do you think we have to drop the support?